neonvm/runner/container-mgr/main.go

```diff
@@ -184,14 +388,48 @@ func (s *cpuServerState) listenForCPUChanges(ctx context.Context, logger *zap.Lo
 	select {
 	case err := <-errChan:
 		if errors.Is(err, http.ErrServerClosed) {
-			logger.Info("cpu_change server closed")
+			logger.Info("http server closed")
 		} else if err != nil {
-			logger.Fatal("cpu_change server exited with error", zap.Error(err))
+			logger.Fatal("http exited with error", zap.Error(err))
 		}
 	case <-ctx.Done():
 		err := server.Shutdown(context.Background())
-		logger.Info("shut down cpu_change server", zap.Error(err))
+		logger.Info("shut down http server", zap.Error(err))
+	}
+}
+
+func handleGetNetworkUsage(logger *zap.Logger, w http.ResponseWriter, r *http.Request, counters *ebpf.Map) {
+	if r.Method != "GET" {
+		logger.Error("unexpected method", zap.String("method", r.Method))
+		w.WriteHeader(400)
+		return
+	}
+
+	var counts vmv1.VirtualMachineNetworkUsage
+	if err := counters.Lookup(NETWORK_INGRESS, &counts.IngressBytes); err != nil {
+		logger.Error("error reading ingress byte counts", zap.Error(err))
+		w.WriteHeader(500)
+		return
+	}
+	if err := counters.Lookup(NETWORK_EGRESS, &counts.EgressBytes); err != nil {
+		logger.Error("error reading egress byte counts", zap.Error(err))
+		w.WriteHeader(500)
+		return
+	}
+	body, err := json.Marshal(counts)
+	if err != nil {
+		logger.Error("could not marshal byte counts", zap.Error(err))
+		w.WriteHeader(500)
+		return
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+	if _, err := w.Write(body); err != nil {
+		logger.Error("could not write body", zap.Error(err))
+		w.WriteHeader(500)
+		return
```

Comment:
I _think_ writing the body preemptively writes the header already (which I guess makes sense if you think about the structure of an HTTP request/response).

Elsewhere, we just ignore the error — if writing the response fails, there's not much we can do.

Reply:
Sounds good, I switched it to how handleCPUCurrent does it